File: TypeScript/src/hero.ts
import { Item } from "./item.js";

export class Hero {

  public name: string;

  public mainhand: Item;

  public offhand: Item;

  public constructor(name: string, mainhand: Item, offhand: Item) {
    this.name = name;
    this.mainhand = mainhand;
    this.offhand = offhand;
  }

  public toString(this: Hero) {
    return `${this.name} (${this.mainhand}, ${this.offhand})`;
  }

  public lmb(this: Hero) {
    return `${this.name} uses ${this.mainhand}!`;
  }

  public rmb(this: Hero) {
    return `${this.name} uses ${this.offhand}!`;
  }
}

File: c++/hero.cpp
#pragma once
#include <iostream>
//#include "hero.h"
#include "item.h"

class Hero {
public:
	char* name;
	Item* mainhand;
	Item* offhand;

  Hero(char* name, Item* mainhand, Item* offhand) {
    this->name = name;
    this->mainhand = mainhand;
    this->offhand = offhand;
  }

  void lmb() {
    std::cout << this->name <<  u8" uses " << this->offhand->name <<  u8"!" << std::endl;
  }

  void rmb() {
    std::cout << this->name <<  u8" uses " << this->mainhand->name <<  u8"!" << std::endl;
  }

  void to_string() {
	  std::cout <<  u8"\n" << this->name <<  u8": {\n  mainhand: " << this->mainhand->name <<  u8",\n  offhand: " << this->mainhand->name <<  u8",\n}" << std::endl;
  }
};
